config/hotreloadable.go

```diff
@@ -0,0 +1,257 @@
+package config
+
+import "time"
+
+// RegisterIntConfigVariable registers int config variable
+func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, keys ...string) {
+	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, keys...)
+}
```

Comment:
Given that this will be shared across several projects I think [this approach](https://github.com/rudderlabs/rudder-server/pull/3034) should be considered.

Reply:
Switching from plain primitives to `Atomic[T]` is not a small task for rudder-server & the idea is to switch to the common library without delays. Such a change could be introduced later and in phases if needed.

On a personal note, if we were to actually refactor in a backwards incompatible manner how reloadable configuration works I would refrain from exposing an API where the caller needs to pass a pointer to `Atomic[T]` and have the Config struct returning a `type Reloadable[T] interface { Get() T }` which always will return a copy and can take care of any synchronisation issues. But this would require effort from rudder-server to adapt.

Reply:
I'm not saying to migrate RS. My approach keeps both but new projects will see the race one as deprecated and start using the atomic one.

Reply:
I would prefer to defer this discussion. The idea behind marking something as deprecated is to eventually switch over to the new one and delete the old in a future release. As mentioned, I have another view with respect to reloadable config api design & I don't expect us to sort this out in this pr 😃